core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -753,21 +749,19 @@ class HttpGoogleServicesDAO(
     } )
   }
 
-  override def getGenomicsOperation(jobId: String): Future[Option[JsObject]] = {
-
-    import spray.json._
-
+  override def getGenomicsOperation(jobId: String): Future[JsObject] = {
     val opId = s"operations/$jobId"
 
     val genomicsApi = new Genomics.Builder(httpTransport, jsonFactory, getGenomicsServiceAccountCredential).setApplicationName(appName).build()
     val operationRequest = genomicsApi.operations().get(opId)
 
-    retryWithRecoverWhen500orGoogleError[Option[JsObject]](() => {
+    retryWithRecoverWhen500orGoogleError[JsObject](() => {
       // Google library returns a Map[String,AnyRef], but we don't care about understanding the response
       // So, use Google's functionality to get the json string, then parse it back into a generic json object
-      Some(executeGoogleRequest(operationRequest).toPrettyString.parseJson.asJsObject)
-    }) {
-      case t: HttpResponseException if t.getStatusCode == StatusCodes.NotFound.intValue => None
+      executeGoogleRequest(operationRequest).toPrettyString.parseJson.asJsObject
+    }) { case e: GoogleJsonResponseException =>
+      val statusCode = StatusCodes.getForKey(e.getStatusCode).getOrElse(StatusCodes.InternalServerError)
```

Comment:
This is a significant change to the previous behaviour. Before, non-404s would potentially get retried because they'd fall through and get caught in `GoogleUtilities.when500orGoogleError`. Now we're just passing the Google error straight up the chain, wrapped in a `RawlsExceptionWithErrorReport`, so it bypasses our retry mechanism.

I'd guess this is probably why the `Option` was there -- to distinguish between a 404-failure (which just means we couldn't find your job, turns into `None`) and a retry-able failure (might succeed next time, or otherwise return a properly-failed `Future` if it's legit buggered).